OBSDOCS#1324: Improve 'troubleshooting monitoring issues: Investigati… (#91118)
openshift/openshift-docs: modules/monitoring-investigating-why-user-defined-metrics-are-unavailable.adoc

ifdef::openshift-rosa,openshift-rosa-hcp,openshift-dedicated[]
* You have access to the cluster as a user with the `dedicated-admin` role.
endif::openshift-rosa,openshift-rosa-hcp,openshift-dedicated[]
* You have installed the {oc-first}.
* You have enabled and configured monitoring for user-defined projects.
* You have created a `ServiceMonitor` resource.

.Procedure

. Ensure that your project is not excluded from user workload monitoring. The following examples use the `ns1` project.

.. Verify that the project _does not_ have the `openshift.io/user-monitoring=false` label attached:
+
[source,terminal]
----
$ oc get namespace ns1 --show-labels | grep 'openshift.io/user-monitoring=false'

> ----

Author: Hello @juzhao I came up with this command to check, is there maybe a better option?

Reviewer: /lgtm
NOTE: default label for user project is openshift.io/user-monitoring=true, but you won't see the label for it if don't set it explicitly.

ifdef::openshift-rosa,openshift-rosa-hcp,openshift-dedicated[]
* You have access to the cluster as a user with the `dedicated-admin` role.
endif::openshift-rosa,openshift-rosa-hcp,openshift-dedicated[]
* You have installed the {oc-first}.
* You have enabled and configured monitoring for user-defined projects.
* You have created a `ServiceMonitor` resource.

.Procedure

> . Ensure that your project is not excluded from user workload monitoring. The following examples use the `ns1` project.

Author: There are more than one example that it refers to here, so I think keeping it as is, is more accurate, thanks!

.. Verify that the project _does not_ have the `openshift.io/user-monitoring=false` label attached:
+
[source,terminal]
----
$ oc get namespace ns1 --show-labels | grep 'openshift.io/user-monitoring=false'
----
+
[NOTE]
====

> The default label for user workload projects is `openshift.io/user-monitoring=true`. However, the label is not visible unless you set it explicitly.

Reviewer 2: ```suggestion
The default label for a user's workload project is `openshift.io/user-monitoring=true`. However, the label is not visible unless you set it explicitly.
```

Reviewer 2: These two sentences are a little confusing. Isn't a default setting by definition set so the user wouldn't have to explicitly set it.

Author: The label is set, but it is not visible to users unless they explicitly set it. I will think of how to rewrite

====

> .. If the label is attached, remove the label:
> +
> .Example of removing the label from the project

Reviewer 3: ```suggestion
.. If the label is attached, remove the label from the project:
+
```

Not sure if "by entering/running the following command" is used throughout the module or not -- you might want to use it consistently in this file. 
Guideline https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#procedure-step-that-introduces-a-command

[source,terminal]
----
$ oc label namespace ns1 'openshift.io/user-monitoring-'
----
+
.Example output
[source,terminal]
----
namespace/ns1 unlabeled
----

> . Check that the corresponding labels match in the service and `ServiceMonitor` resource configurations. The following examples use the `prometheus-example-app` service, the `prometheus-example-monitor` service monitor, and the `ns1` project.

Reviewer 2: ```suggestion
. Check that the corresponding labels match in the service and `ServiceMonitor` resource configuration. The following example uses the `prometheus-example-app` service, the `prometheus-example-monitor` service monitor, and the `ns1` project.
```

Author: I will change the configuration, but again, the plural for examples makes more sense to me here

Reviewer 2: If service is a configuration it should be in backticks.

> .. Obtain the label defined in the service.

Reviewer 3: ```suggestion
.. Obtain the label defined in the service by running the following command:
```

This comment is optional; but just in case you want to update this sentence per https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#procedure-step-that-introduces-a-command

Author: Oh yes, I did not want to add this due to consistency. None of the monitoring docs have this anywhere, I am thinking of eventually creating an issue for this to have it unified :+1:

    app: prometheus-example-app
----
+

> .. Check that the `matchLabels` definition in the `ServiceMonitor` resource configuration matches the label output in the preceding step.

Reviewer 3: ```suggestion
.. Check that the `matchLabels` definition in the `ServiceMonitor` resource configuration matches the label output in the previous step.
```

IMO - _previous_ is widely/commonly used. This is up to you

Author: I will include this in the next PR.

Reviewer 3: Thanks!!
